python/cuml/experimental/explainer/kernel_shap.pyx

```diff
@@ -0,0 +1,505 @@
+        cuML's kernel SHAP supports tabular data for now, so it expects
+        a background dataset, as opposed to a shap.masker object. To respect
+        a hierarchical structure of the data, use the (temporary) parameter
+        'masker_type'
+        Acceptable formats: CUDA array interface compliant objects like
+        CuPy, cuDF DataFrame/Series, NumPy ndarray and Pandas
+        DataFrame/Series.
+    nsamples : int
+        Number of samples to use to estimate shap values.
+    masker_type: {'independent', 'partition'} default = 'independent'
+        If 'independent' is used, then this is equivalent to SHAP's
+        independent masker and the algorithm is fully GPU accelerated.
+        If 'partition' then it is equivalent to SHAP's Partition masker,
+        which respects a hierarchical structure in the background data.
+    link : function or str
+        The link function used to map between the output units of the
+        model and the SHAP value units.
+    random_state: int, RandomState instance or None (default)
+        Seed for the random number generator for dataset creation.
+    gpu_model : bool
+
+    handle : cuml.raft.common.handle
+        Specifies the handle that holds internal CUDA state for
+        computations in this model. Most importantly, this specifies the CUDA
+        stream that will be used for the model's computations, so users can
+        run different models concurrently in different streams by creating
+        handles in several streams.
+        If it is None, a new one is created.
+    dtype : np.float32 or np.float64 (default=None)
+        Parameter to specify the precision of data to generate to call the
+        model. If not specified, the explainer will try to get the dtype
+        of the model, if it cannot be queried, then it will defaul to
+        np.float32.
+    output_type : 'cupy' or 'numpy' (default:None)
+        Parameter to specify the type of data to output.
+        If not specified, the explainer will try to see if model is gpu based,
+        if so it will default to `cupy`, otherwise it will default to `numpy`.
+        For compatibility with SHAP's graphing libraries, specify `numpy`.
+
+    """
+
+    def __init__(self,
+                 model,
+                 data,
+                 nsamples=None,
+                 link='identity',
+                 verbosity=False,
+                 random_state=None,
+                 gpu_model=None,
+                 handle=None,
+                 dtype=None,
+                 output_type=None):
+
+        self.handle = Handle() if handle is None else handle
+
+        self.link = link
+        self.link_fn = get_link_fn_from_str(link)
+        self.model = model
+        self.order = get_tag_from_model_func(func=model,
+                                             tag='preferred_input_order',
+                                             default='C')
+        if gpu_model is None:
```

Comment:
Indeed, since there are quite a few things that are deduced from `model` to enhance the working of the Explainer, I think implementing custom `get_param/set_param` for this not Estimator classes might be the way to go, what do you think?